### PR TITLE
Add vocoder modules related to VITS 

### PR DESCRIPTION
**Pull request by kan-bayashi**
Subset of #3423 

This PR adds vocoder related modules, which used for VITS.
In the next PR, I will add more modules.

### CODE REVIEW

File: espnet2/gan_tts/vits/residual_block.py [hunk 1]
```diff
@@ -0,0 +1,247 @@
...
+        if global_channels > 0:
+            self.conv1x1_glo = Conv1d1x1(global_channels, gate_channels, bias=False)
+        else:
+            self.conv1x1_glo = None
+
+        # conv output is split into two groups
+        gate_out_channels = gate_channels // 2
+        self.conv1x1_out = Conv1d1x1(gate_out_channels, residual_channels, bias=bias)
+        if skip_channels > 0:
+            self.conv1x1_skip = Conv1d1x1(gate_out_channels, skip_channels, bias=bias)
```
Comment (r9y9) [inline espnet2/gan_tts/vits/residual_block.py:112]: You can merge these two conv layers into single one for efficiency (there should not be numerical difference though), as done in https://github.com/jaywalnut310/vits/blob/2e561ba58618d021b5b8323d3765880f7e0ecfdb/modules.py#L144.

Ref also: https://github.com/NVIDIA/waveglow/pull/20

My own PWG implementation also adapts this trick.

Comment (r9y9) [reply espnet2/gan_tts/vits/residual_block.py:112]: assuming skip_channels > 0

File: espnet2/gan_tts/vits/residual_block.py [hunk 2]
```diff
@@ -0,0 +1,247 @@
...
+
+    def __init__(
+        self,
+        kernel_size: int = 3,
+        residual_channels: int = 64,
+        gate_channels: int = 128,
+        skip_channels: int = 64,
+        aux_channels: int = 80,
+        global_channels: int = -1,
+        dropout_rate: float = 0.0,
```
[the remark this answers is not in the document]
Comment (r9y9) [reply espnet2/gan_tts/vits/residual_block.py:57]: It is okay to keep it here for compatibility with ParallelWaveGAN repo or VITS hyper parameters.

**Review by r9y9 — COMMENTED** [review]
Sorry for the late review. Added comments!

File: espnet2/gan_tts/vits/residual_block.py (hunk 1 above)
Comment (kan-bayashi) [reply espnet2/gan_tts/vits/residual_block.py:112]: Oh, maybe I misunderstood the original code.
I thought that they did not use skip connection part so I added skip channels < 1 case.
But that is wrong. I will fix this part.

File: espnet2/gan_tts/vits/residual_block.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (kan-bayashi) [reply espnet2/gan_tts/vits/residual_block.py:57]: Thank you for your suggestion.
I want to keep it as is for now.

**Comment by kan-bayashi** [comment]
@r9y9 I can fix the a bug about the WaveNet part :)
I refactored it, could you check again?

File: espnet2/gan_tts/vits/wavenet.py [hunk 3]
```diff
@@ -0,0 +1,196 @@
...
+        gate_channels: int = 128,
+        skip_channels: int = 64,
+        global_channels: int = -1,
+        dropout_rate: float = 0.0,
+        bias: bool = True,
+        use_weight_norm: bool = True,
+        use_first_conv: bool = False,
+        use_last_conv: bool = False,
+        scale_residual: bool = False,
+        scale_skip_connect: bool = False,
```
Comment (kan-bayashi) [inline espnet2/gan_tts/vits/wavenet.py:43]: @r9y9 How do you think about default value of these parameters?

Comment (r9y9) [reply espnet2/gan_tts/vits/wavenet.py:43]: This is reasonable to me. IIRC, there weren't significant differences in my past experiments.

**Review by r9y9 — APPROVED** [review]
LGTM, except for my one minor suggestion to make the code cleaner. Thanks a lot for your work!

File: espnet2/gan_tts/vits/residual_block.py [hunk 4]
```diff
@@ -151,13 +151,16 @@ def forward(
 
         x = torch.tanh(xa) * torch.sigmoid(xb)
 
-        # for skip connection
+        # residual + skip 1x1 conv
+        x = self.conv1x1_out(x)
+
         s = None
         if self.skip_channels > 0:
```
[the remark this answers is not in the document]
Comment (kan-bayashi) [reply espnet2/gan_tts/vits/residual_block.py:165]: Agree. I will remove it.

**Pull request merged**